### PR TITLE
Added diagnostic for @FacesValidator classes that do not implement ja…

### PR DESCRIPTION
**Pull request by sajeerzeji**
Fixes https://github.com/eclipse-lsp4jakarta/lsp4jakarta/issues/773

Implements a diagnostic that detects classes annotated with @FacesValidator that do not implement jakarta.faces.validator.Validator.

https://github.com/user-attachments/assets/2195273c-0ee1-43f0-a201-ee1cd7c321ae

### CODE REVIEW

**Review by Rejoice472 — COMMENTED** [review]
The build is failing due to a compilation error in JakartaFacesTest.java at line 56. The error message indicates:

Code
error: cannot find symbol
  symbol:   variable d
  location: class JakartaFacesTest

**Comment by sajeerzeji** [comment]
> The build is failing due to a compilation error in JakartaFacesTest.java at line 56. The error message indicates:
> 
> Code error: cannot find symbol symbol: variable d location: class JakartaFacesTest

It was a test variable renaming mismatch, just fixed it

File: src/main/resources/io/openliberty/tools/intellij/lsp4jakarta/messages/messages.properties [hunk 1]
```diff
@@ -256,6 +256,9 @@ ServerEndpointDuplicateVar = Server endpoint paths must not use the same variabl
 publicNoArgConstructorMissing = WebSocket endpoint class {0} must declare a public no-argument constructor.
 DuplicateLifeCycleAnnotation = Life cycle annotation {0} already registered with another method.
 
+# FacesDiagnosticsCollector
+FacesValidatorMustImplement = Annotated classes with @FacesValidator must implement the jakarta.faces.validator.Validator interface.
```
Comment (Joseph-Bineesh) [inline src/main/resources/io/openliberty/tools/intellij/lsp4jakarta/messages/messages.properties]: Can we rephrase to 
'Classes annotated with @FacesValidator must implement the jakarta.faces.validator.Validator interface.'

**Review by Joseph-Bineesh — CHANGES_REQUESTED** [review]

File: src/main/resources/io/openliberty/tools/intellij/lsp4jakarta/messages/messages.properties (hunk 1 above)
Comment (sajeerzeji) [reply src/main/resources/io/openliberty/tools/intellij/lsp4jakarta/messages/messages.properties]: Updated in both lsp4jakarta and LTI

**Review by Joseph-Bineesh — DISMISSED** [review]
LGTM

**Review by Rejoice472 — DISMISSED** [review]

File: src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/faces/JakartaFacesTest.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import java.io.File;
+import java.util.Arrays;
+
+import static io.openliberty.tools.intellij.lsp4jakarta.it.core.JakartaForJavaAssert.*;
+
+@RunWith(JUnit4.class)
+public class JakartaFacesTest extends BaseJakartaTest {
```
Comment (venmanyarun) [inline src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/faces/JakartaFacesTest.java:36]: Consider edge cases

- A class annotated with @FacesValidator that extends a superclass which implements Validator (inherited implementation) — is doesImplementInterfaces recursive?
- A @FacesValidator applied to an abstract class — should it be exempt?
- A class implementing a raw Validator (without the generic type)

File: src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/faces/JakartaFacesTest.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        VirtualFile javaFile = LocalFileSystem.getInstance().refreshAndFindFileByPath(ModuleUtilCore.getModuleDirPath(module)
+                + "/src/main/java/io/openliberty/sample/jakarta/faces/DontImplementValidator.java");
+        String uri = VfsUtilCore.virtualToIoFile(javaFile).toURI().toString();
+
+        JakartaJavaDiagnosticsParams diagnosticsParams = new JakartaJavaDiagnosticsParams();
+        diagnosticsParams.setUris(Arrays.asList(uri));
+
+        // expected: diagnostic on class name (line 5, col 13..35 — "DontImplementValidator")
+        Diagnostic expecteDiagnostic = d(5, 13, 35,
```
Comment (venmanyarun) [inline src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/faces/JakartaFacesTest.java]: typo

**Review by venmanyarun — CHANGES_REQUESTED** [review]

File: src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/faces/JakartaFacesTest.java (hunk 2 above)
Comment (sajeerzeji) [reply src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/faces/JakartaFacesTest.java:36]: From this, the first case was missing to handle.
1. **"A class annotated with @FacesValidator that extends a superclass which implements Validator":** This is handled and tests added.
2. **"A @FacesValidator applied to an abstract class":** This was already been handled, tests added now.
3. **"A class implementing a raw Validator":** This was also already been handled, tests added now.

File: src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/faces/JakartaFacesTest.java (hunk 3 above)
Comment (sajeerzeji) [reply src/test/java/io/openliberty/tools/intellij/lsp4jakarta/it/faces/JakartaFacesTest.java]: This is fixed

**Review by venmanyarun — APPROVED** [review]

**Review by Rejoice472 — APPROVED** [review]

**Review by Joseph-Bineesh — APPROVED** [review]

**Review by turkeylurkey — APPROVED** [review]
Looks good
